### PR TITLE
P4 1090 send azure events

### PR DESCRIPTION
**Pull request by starswan**
Fixes P4-1090

## Proposed Changes

- use a new base image for the app, as we were using an outdated MoJ image. Activate AppInsights on staging, preprod and production

## To test/demo change

- Browse environment, and check matching Azure App Insights data

### CODE REVIEW

**Comment by alexdesi** [comment]
Broken tests here.
@starswan how we can test this?
since it changes Circle CI it could be cause of potential issues on today's deploy (12th march),
it might be wise to merge it later.

**Comment by alexdesi** [comment]
By the way, I am not advocating *Fat Models and skinny Controllers*, but fat controller it might be even worst.

**Comment by starswan** [comment]
> Broken tests here.
> @starswan how we can test this?
> since it changes Circle CI it could be cause of potential issues on today's deploy (12th march),
> it might be wise to merge it later.

Yes not planning to release this today. The only test is to run it on staging - but it doesn't change much on a fundamental level

File: Dockerfile [hunk 1]
```diff
@@ -27,12 +30,23 @@ RUN curl -sL https://deb.nodesource.com/setup_13.x | bash - \
    && apt-get clean
 
 RUN yarn install
+
 # Have to set SECRET_KEY_BASE here to arbitrary string, otherwise task doesn't run
-RUN SECRET_KEY_BASE=valuenotactuallyused rails assets:precompile
+RUN SECRET_KEY_BASE=valuenotactuallyused bundle exec rails assets:precompile
+
+# Run the application as user `moj` (created in the base image)
```
Comment (martyn-w) [inline Dockerfile]: is this comment still relevant given that the base image has changed from `ministryofjustice/ruby:2.6.2-webapp-onbuild` to `ruby:2.6.2-stretch`
...and on a related note, what is the reason for changing the base image?

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,6 +21,14 @@ aliases:
...
       branches:
         only: master
+  - &only_master_tags_and_p4_1090
+    filters:
+      tags:
+        only: /.*/
+      branches:
+        only:
+        - master
+        - p4-1090-send-azure-events
```
Comment (martyn-w) [inline .circleci/config.yml]: this seems a bit clunky: a comment might help explain why only  this branch and master should be deployed?

File: Dockerfile (hunk 1 above)
Comment (starswan) [reply Dockerfile]: Yes the comment is wrong. The base image didn't work, and turned out to be deprecated and undocumented

File: .circleci/config.yml (hunk 2 above)
Comment (starswan) [reply .circleci/config.yml]: Its because I wanted to test deployments.

**Comment by martyn-w** [comment]
ok, looks good - just need to get tests to pass 👍

**Pull request merged**